active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,199 @@
+
+## Is this the right control? 
+
+Use a **NumberBox** to capture and display mathematical input. A **NumberBox** can also be enabled with basic calculator support to compute multiplication, division, addition, and subtraction across parenthetical order with standard operator precedence.
+
+## Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a simple NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+![](images/numberbox-placeholder-text.png)
+
+### Enable calculation support
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+![](images/numberbox-acceptscalculation.png)
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for items in order" 
+    StepSize="0.1"
+    UpDownButtonsEnabled="True"
+    HyperDragEnabled="True" 
+    HyperScrollEnabled="True" />
+```
+
+![](images/numberbox-updownbuttons.png)
+
+### Format display and output
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for solution components" 
+    Header="Enter a percentage:"
+    PlaceholderText="2/3" 
+    MinValue="0"
+    MaxValue="100" 
+    AreLeadingZerosTrimmed="False"
+    DecimalPrecision="5"
+    DoesInputRound=False" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+runtimeclass NumberBoxValueChangedEventArgs
```

Comment:
For consistency with other "...ChangedEventArgs" it would be better (and more helpful) if this class included three properties: `CurrentText`, `OldValue`, and `NewValue`.
The names `Text` and `Value` on their own are ambiguous.

`CurrentText`, and `NewValue` should be the same as accessing the properties directly from the sender but are provided as a convenience to save on the cost of looking them up. `CurrentText` rather than `Text` to avoid confusion with the property and to make it clear how it compares to the other two properties.

Having access to both the old and new values can be useful when tracking changes between valid and invalid states as would be necessary as part of custom validation logic.

I don't feel it's necessary to have access to `OldText` as it may not necessarily correspond to `OldValue` unless the two were tracked in sync. Anyway it's easy to reverse engineer the old text from the old value. If there is a good reason for needing `OldText` I'd be interested it hear it--it's just not something I've ever needed or something I can imagine a case for needing.